public/scripts.js

```diff
@@ -41,5 +66,50 @@ document.addEventListener('drop', (event) => {
     fileInput.files = event.dataTransfer.files;
 
     // Submit form
-    fileInput.closest('form').submit();
+    fileInput.closest('form').requestSubmit();
+});
+
+elements.uploadForm.addEventListener('submit', (event) => {
+    event.preventDefault();
+    console.log("Submit event:");
+    console.log(event);
+
+    var xhr = new XMLHttpRequest();
+    xhr.upload.addEventListener('loadstart', (event) => {
+        uploadStart(event.total);
+    }, false);
+    xhr.upload.addEventListener('progress', (event) => {
+        uploadProgress(event.loaded);
+    }, false);
+    xhr.upload.addEventListener('load', (event) => {
+        uploadProgress(event.total);
+    }, false);
+
+    xhr.addEventListener('readystatechange', (event) => {
+        console.log(event.target.readyState);
+        if (event.target.readyState === XMLHttpRequest.DONE) {
+            const status = event.target.status;
+            if (status === 0 || (status >= 200 && status < 400) && event.target.responseText) {
+                console.log('readystatechange', event);
+                //we got a response from the server and we're replacing the whole current document content with it, simulating a page reload
+                var newDocument = document.open();
+                newDocument.write(event.target.responseText);
+                newDocument.close();
```

Comment:
So this doesn't really reload the page, so we have conflicts with the JS executing again. We could solve it by just manually displaying the success banner from JS.

Reply:
@dbrgn What do you think? I'd hate the duplication of rendering the success message in JS as well, but it is the only way I can think of currently.

Reply:
I found a better solution by wrapping the whole JS in an `addEventListener("DOMContentLoaded"`. That way the constants are local to the event handler and they re-attach again to the DOM elements after loading the page:
https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event